docs/DeveloperGuide.md

```diff
@@ -1,5 +1,26 @@
 # Developer Guide
 
```

Comment:
Better try to simplify the sentences, and use more hierarchy within a section of code instead a chunk of code to improve readability.